test/e2e/gameserver_test.go

```diff
@@ -320,7 +336,7 @@ func TestGameServerRestartBeforeReadyCrash(t *testing.T) {
 		}
 		return false
 	})
-	if err != nil {
+	if !assert.NoError(t, err) {
```

Comment:
So we've got a consistent pattern of doing this as established i.e. check if err != nil and then failing with  t.Fatalf or assert.Failx()

What's the value add of changing this here? assert.NoError() doesn't seem like it's doing anything extra here?